### ISSUE TITLE
ArrayIndexOutOfBoundsException in Pitest on Checkstyle Project

### ISSUE THREAD

**Issue by rnveach** [hcoles/pitest#302]
Project: https://github.com/checkstyle/checkstyle

We use to be able to run pitest just fine on the project around 4 months ago, but I tried running it again today on our latest code and it gave an ArrayIndexOutOfBoundsException.

````
mvn clean compile test org.pitest:pitest-maven:mutationCoverage

Apache Maven 3.2.1 (ea8b2b07643dbb1b84b6d16e1f08391b666bc1e9; 2014-02-14T12:37:52-05:00)
Maven home: D:\Program Files\apache-maven-3.2.1
Java version: 1.8.0_77, vendor: Oracle Corporation
Java home: D:\Program Files\JDK8\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 7", version: "6.1", arch: "x86", family: "dos"

Results :

Tests run: 1865, Failures: 0, Errors: 0, Skipped: 1

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:25 min (Wall Clock)
[INFO] Finished at: 2016-10-03T10:04:26-05:00
[INFO] Final Memory: 33M/182M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.pitest:pitest-maven:1.1.10:mutationCoverage (default-cli) on project checkstyle: Execution default-cli of goal org.pitest:pitest-maven:1.1.10:mutationCoverage failed: 65535

Caused by: org.pitest.util.PitError: 65535

Please copy and paste the information and the complete stacktrace below when reporting an issue
VM : Java HotSpot(TM) Client VM
Vendor : Oracle Corporation
Version : 25.77-b03
Uptime : 206209
Input -> 
 1 : -Dmaven.home=D:\Program Files\apache-maven-3.2.1
 2 : -Dclassworlds.conf=D:\Rickys\checkstyleWorkspaceEclipse\.metadata\.plugins\org.eclipse.m2e.launching\launches\m2conf6109215438002811224.tmp
 3 : -Dfile.encoding=Cp1252
BootClassPathSupported : true

	at org.pitest.util.Unchecked.translateCheckedException(Unchecked.java:20)
	at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:103)
	at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:49)
	at org.pitest.mutationtest.tooling.MutationCoverage.runReport(MutationCoverage.java:110)
	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:103)
	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:45)
	at org.pitest.maven.RunPitStrategy.execute(RunPitStrategy.java:35)
	at org.pitest.maven.AbstractPitMojo.analyse(AbstractPitMojo.java:398)
	at org.pitest.maven.AbstractPitMojo.execute(AbstractPitMojo.java:342)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:133)
	... 12 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: 65535
	at org.pitest.reloc.asm.ClassReader.readLabel(Unknown Source)
	at org.pitest.reloc.asm.ClassReader.a(Unknown Source)
	at org.pitest.reloc.asm.ClassReader.a(Unknown Source)
	at org.pitest.reloc.asm.ClassReader.b(Unknown Source)
	at org.pitest.reloc.asm.ClassReader.accept(Unknown Source)
	at org.pitest.reloc.asm.ClassReader.accept(Unknown Source)
	at org.pitest.classinfo.ClassInfoVisitor.getClassInfo(ClassInfoVisitor.java:46)
	at org.pitest.classinfo.Repository.nameToClassInfo(Repository.java:70)
	at org.pitest.classinfo.Repository.fetchClass(Repository.java:60)
	at org.pitest.classinfo.NameToClassInfo.apply(NameToClassInfo.java:16)
	at org.pitest.classinfo.NameToClassInfo.apply(NameToClassInfo.java:6)
	at org.pitest.functional.FCollection.flatMapTo(FCollection.java:56)
	at org.pitest.functional.FCollection.flatMap(FCollection.java:66)
	at org.pitest.classpath.CodeSource.getTests(CodeSource.java:58)
	at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:80)
	... 20 more
````

This may be related to #297 as we recently upgraded to Java 8, but we don't use gradle.

**Comment by KyleRogers** [comment]
I think this is a duplicate of #297 --  original cause is a bug in jdk compiler and this this is not related to gradle. You can use eclipse compiler which seems not to be affected.

The JDK bug is already open for a year and there seems to be no progress/work on this topic. ASM folks seem to have closed the bug, so I don't expect any fix from there.

Working around the bug on the pitest end requires forking ASM.

**Comment by rnveach** [comment]
@KyleRogers 
Thanks. ECJ does work for our project but only after removing that resources be compiled.
ECJ seems to print out compile errors when included in the maven build that the normal JDK doesn't.

**Comment by rnveach** [comment]
Is it possible to update pitest to atleast display the file that is the problem so it is easier to add it to the exclude list?
